FSJ_django20_project/FSJ/models_student.py

```diff
@@ -1 +1 @@
-from django.db import models
from .models_FSJUser import FSJUser
from django.utils.translation import gettext_lazy as _
from django import forms

# This class inherits from a standard FSJ User and extends for Student specific attributes and methods
class Student(FSJUser):
    FIRST = 1
    SECOND = 2
    THIRD = 3
    FOURTH = 4
    FIFTH = 5
    YEAR_CHOICES = (
        (FIRST, _('First')),
        (SECOND, _('Second')),
        (THIRD, _('Third')),
        (FOURTH, _('Fourth')),
        (FIFTH, _('Fifth')),
    )
    
    program = models.CharField(max_length = 255, verbose_name = _("Program"))
    year = models.IntegerField(blank = False, choices = YEAR_CHOICES, default = FIRST, verbose_name = _("Year of Study"))

    def user_class(self):
        return "Student"
            
    def get_program(self):
        return self.program
    
    def student_id(self):
        return self.userid
    
\ No newline at end of file
+from django.db import models
from .models_FSJUser import FSJUser
from django.utils.translation import gettext_lazy as _
from django import forms
from .models_yearofstudy import *

# This class inherits from a standard FSJ User and extends for Student specific attributes and methods
class Student(FSJUser):

    program = models.CharField(max_length = 255, verbose_name = _("Program"))
    year = models.OneToOneField(YearOfStudy, on_delete=models.CASCADE)

    def user_class(self):
        return "Student"
            
    def get_program(self):
        return self.program
    
    def student_id(self):
        return self.userid
    
```

Comment:
one to one means only one student may be in, say, second year

Reply:
Do we really want to cascade on delete? I can't see us deleting a year from existence by deleting a student, but likewise can't see us deleting a student when a year gets deleted that they're associated with (perhaps prevent deletion of the year, or perhaps set all students with that year to some default value or allow the value to be null...)